workspace/manager.py

```
@@ -240,6 +240,105 @@ def create_workspace(
+        attempt to inject the token into an HTTPS URL for auth when provided.
+        """
+        if not repo_url or not isinstance(repo_url, str):
+            return {"ok": False, "error": "no_repo_url"}
+        try:
+            import subprocess
+            env = dict(**os.environ)
+            env.setdefault("GIT_TERMINAL_PROMPT", "0")
+            auth_url = repo_url
+            if token and repo_url.startswith("https://"):
+                auth_url = repo_url.replace("https://", f"https://{token}@")
+            proc = subprocess.run(["git", "ls-remote", "--heads", auth_url], stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env, timeout=timeout)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)